### PR TITLE
Add support for Selinux mount context labels

### PR DESCRIPTION
**Pull request by mrunalp**
Signed-off-by: Mrunal Patel <mrunalp@gmail.com>

### CODE REVIEW

**Comment by mrunalp** [comment]
@crosbymichael @LK4D4 ping
cc: @rhatdan

**Comment by crosbymichael** [comment]
Why does this not get placed in the Mount.Options array?

**Comment by mrunalp** [comment]
@crosbymichael This is for convenience. We could pass it to individual mounts as well. (We had a MountLabel in libcontainer configuration).

**Comment by mrunalp** [comment]
An argument in favor is that the value of the label is the same for all the mount points (that we need to label).

**Comment by crosbymichael** [comment]
Could it every need to be different or will it **always** be the same?

**Comment by mrunalp** [comment]
AFAIK it wouldn't be but would defer to @rhatdan on that.

File: config-linux.md [hunk 1]
```diff
@@ -545,6 +545,18 @@ Its value is either slave, private, or shared.
...
 ```
 
+## Mount Label
+
+`mountLabel` will set the Selinux context for the mounts in the container.
+
+###### Example
+
+```json
+    "mountLabel": [
```
Comment (crosbymichael) [inline config-linux.md]: This should not be an array `[]` right?

**Comment by crosbymichael** [comment]
ok, looking at the existing code this looks good.  Only thing is that your json example has this as an array and not just a string

**Comment by mrunalp** [comment]
@crosbymichael Fixed.

**Comment by crosbymichael** [comment]
LGTM

**Comment by rhatdan** [comment]
Right now we are still always planning on labeling content inside of the container with the same label.  
LGTM

**Comment by vbatts** [comment]
@mrunalp would you mind adding it to the JSON schema too? otherwise I can do a follow-up PR.

**Comment by vbatts** [comment]
LGTM based on your response ^^

**Comment by mrunalp** [comment]
I'll add it.

**Comment by mrunalp** [comment]
@vbatts Added to schema.

File: config-linux.md [hunk 2]
```diff
@@ -545,6 +545,16 @@ Its value is either slave, private, or shared.
...
 ```
 
+## Mount Label
+
+`mountLabel` will set the Selinux context for the mounts in the container.
+
+###### Example
+
+```json
+    "mountLabel": "system_u:object_r:svirt_sandbox_file_t:s0:c715,c811"
```
Comment (vbatts) [inline config-linux.md]: sorry. One more nit. Please add this line to the "complete json" example in the config.md.

File: schema/schema-linux.json [hunk 3]
```diff
@@ -330,6 +330,10 @@
             "readonlyPaths": {
                 "id": "https://opencontainers.org/schema/bundle/linux/readonlyPaths",
                 "$ref": "defs.json#/definitions/ArrayOfStrings"
+            },
+            "mountLabel": {
+                "id": "https://opencontainers.org/schema/bundle/linux/mountLabel",
+                "type": "string"
```
Comment (vbatts) [inline schema/schema-linux.json]: cool

File: config-linux.md (hunk 2 above)
Comment (mrunalp) [inline config-linux.md]: Good catch. Fixed.

File: config-linux.md [hunk 4]
```diff
@@ -545,6 +545,16 @@ Its value is either slave, private, or shared.
     ]
 ```
 
+## Mount Label
+
+`mountLabel` will set the Selinux context for the mounts in the container.
```
Comment (wking) [inline config-linux.md]: Do we want cross-links between this and config.md's **`process.selinuxLabel`**?  I expect folks interested in one would also be interested in the other.

**Comment by vbatts** [comment]
LGTM

**Pull request merged**